### PR TITLE
Cascadeclassifier

### PR DESCRIPTION
**Pull request by ntorqulu**
will be able to merge?

### CODE REVIEW

**Comment by ntorqulu** [comment]
En aquest pr he inclòs:
1. Model BERT per a la classificació de seqüències, només funciona amb el encoding 4 row
2. Refactor del codi en singlerank models i mulirank model. Col.locat en singlerank tots els models implementats fins are i en multirank els models implementats que entrenen multiples ranks a l'hora. Són MLPs simples, no s'han incorporat els models fets per a singlerank. TODO.
3. Implementat els models multirank hierarchical_model, cascade_hierarchical_model and gnn_hierarchical_model

BREU EXPLICACIÓ DE CADA PUNT:
1. BERT Model Implementation
- BERT model només corre en 4row matrix encoder
- Afegit json file amb els hparams per a executar el model
- Per a testejar-lo:
python run_singlerank_experiment.py --config src/models/hyperparams/singlerank/bert_hparams.json

2.  Folder Restructuring
Organized into Single-Rank and Multi-Rank Models:
- singlerank: Contains hyperparameter configurations for models predicting a single taxonomic rank
- multirank: Contains hyperparameter configurations for hierarchical models that predict multiple ranks simultaneously
Per executar models singlerank, fer-ho com fins ara pero cambiant l'ordre d'execució per:
python run_singlerank_experiment.py --config src/models/hyperparams/singlerank/{fitxer_json}
python run_singlerank_experiment.py --config src/models/hyperparams/singlerank/nanni_cnn1_hparams.json

3. Multi-Rank Models Implementation
Added three distinct hierarchical classification approaches:
- Basic Hierarchical Model: primer feature extractor de la seqüència, i després parallel prediction per als diferents ranks
- Cascade Hierarchical Model: sequencial prediction from kingdom → phylum → class → order. Each level's prediction informs the next level using confidence-weighted inputs. Implements specialized loss functions that enforce taxonomic consistency
- Graph Neural Network Model: Represents taxonomic levels as nodes in a graph network. Graph convolutional layers for bidirectional info flow. Attention mechanism incorporated

Usage for multi-rank:
python run_multirank_experiment.py --config src/models/hyperparams/multirank/cascade_hparams.json
python run_multirank_experiment.py --config src/models/hyperparams/multirank/gnn_hierarchical_hparams.json
python run_multirank_experiment.py --config src/models/hyperparams/multirank/hierarchical_hparams.json

**Comment by ntorqulu** [comment]
I un altre TODO, només s'ha testejat per als ranks kingdom, phylum,  class, order. Encara falta testejar-lo en el all-ranks parquet files
